packages/edit-widgets/src/store/actions.js

```diff
@@ -108,17 +108,7 @@ export function* saveWidgetArea( widgetAreaId ) {
 		( { sidebar } ) => sidebar === widgetAreaId
 	);
 
-	// Remove all duplicate reference widget instances
-	const usedReferenceWidgets = [];
-	const widgetsBlocks = post.blocks.filter( ( { attributes: { id } } ) => {
-		if ( id ) {
-			if ( usedReferenceWidgets.includes( id ) ) {
-				return false;
-			}
-			usedReferenceWidgets.push( id );
-		}
-		return true;
-	} );
+	const widgetsBlocks = post.blocks;
```

Comment:
```js
// Remove all duplicate reference widget instances for legacy widgets.
// Why? We filter out the widgets with duplicate IDs to prevent adding more than one instance of a widget
// implemented using a function. WordPress doesn't support having more than one instance of these, if you try to
// save multiple instances of these in different sidebars you will run into undefined behaviors.
const usedReferenceWidgets = [];
const widgetsBlocks = post.blocks.filter( ( block ) => {
    debugger;
    const { id } = block.attributes;

    if ( block.name === 'core/legacy-widget' && block.attributes?.id ) {
        if ( usedReferenceWidgets.includes( id ) ) {
            return false;
        }
        usedReferenceWidgets.push( id );
    }
    return true;
} );
```

Reply:
I gave this a spin and it seemed to work.

I'd say we'd need some confirmation that the widget `id` guard is still correct as even with Adam's comment I (personally) don't have the full context/confidence that I've done this correctly.

Reply:
https://user-images.githubusercontent.com/444434/123297534-8e168b80-d50f-11eb-86f0-5946d162e442.mov

Reply:
Yes, it works! I tested with file and cover block too! 🎉